crates/biome_grit_patterns/src/errors.rs

```diff
@@ -77,7 +80,116 @@ pub enum CompileError {
+                        diag.message(fmt)
+                    }
+                    None => Ok(()),
+                }
+            }
+            CompileError::ParseSnippetError(error) => {
+                fmt.write_markup(markup! { "Error parsing snippet" })?;
+                match error.diagnostics.first() {
+                    Some(diag) => {
+                        fmt.write_str(": ")?;
+                        diag.message(fmt)
+                    }
+                    None => Ok(()),
+                }
+            }
+            CompileError::MissingSyntaxNode => {
+                fmt.write_markup(markup! { "A syntax node was missing" })
+            }
+            CompileError::DuplicateParameters => {
+                fmt.write_markup(markup! { "Duplicate parameters" })
+            }
+            CompileError::InvalidMetavariableRange(_) => {
+                fmt.write_markup(markup! { "Invalid range for metavariable" })
+            }
+            CompileError::MetavariableNotFound(var) => {
+                fmt.write_markup(markup! { "Metavariable not found: "{{var}} })
+            }
+            CompileError::ReservedMetavariable(var) => {
+                fmt.write_markup(markup! { "Reserved metavariable: "{{var}} })
+            }
+            CompileError::UnsupportedKind(kind) => {
+                fmt.write_markup(markup! { "Unsupported syntax kind ("{{kind}}")" })
+            }
+            CompileError::UnexpectedKind(kind) => {
+                fmt.write_markup(markup! { "Unexpected syntax kind ("{{kind}}")" })
+            }
+            CompileError::UnknownFunctionOrPattern(name) => {
+                fmt.write_markup(markup! { "Unknown function or pattern: "{{name}} })
+            }
+            CompileError::LiteralOutOfRange(value) => {
+                fmt.write_markup(markup! { "Literal value out of range: "{{value}} })
+            }
+            CompileError::MissingPattern => fmt.write_markup(markup! { "Missing pattern" }),
+            CompileError::InvalidBracketedMetavariable => {
+                fmt.write_markup(markup! { "Invalid bracketed metavariable" })
+            }
+            CompileError::FunctionArgument(_) => {
+                fmt.write_markup(markup! { "Invalid function argument" })
+            }
+            CompileError::UnknownFunctionOrPredicate(name) => {
+                fmt.write_markup(markup! { "Unknown function or predicate: "{{name}} })
+            }
+            CompileError::UnknownVariable(var) => {
+                fmt.write_markup(markup! { "Unknown variable: "{{var}} })
+            }
+        }
+    }
+
+    fn location(&self) -> Location<'_> {
+        match self {
+            CompileError::ParsePatternError(error) => error
+                .diagnostics
+                .first()
+                .map(Diagnostic::location)
+                .unwrap_or_default(),
+            _ => Location::default(),
+        }
+    }
```

Comment:
Once we remove the `SerializableDiagnostic`, the implementation of `location` and the introduction of `Default` to `Location` won't be needed.

Reply:
I'm not sure I fully understand this one. Do you mean: if it wouldn't contain a vector of diagnostics anymore?

Even then I'm not sure why `location()` wouldn't be needed anymore?

Reply:
Nevermind. I looked at how `Location` is implemented, and all its fields are `Option`, so I suppose `Default` makes sense.